### PR TITLE
Fix for material alpha mapping

### PR DESCRIPTION
**Pull request by jdent02**

### CODE REVIEW

**Comment by jdent02** [comment]
#168

File: projectwriter.py [hunk 1]
```diff
@@ -2221,6 +2221,7 @@ def __emit_texture_instance(self, texture, texture_name, bump_bool, node=None, m
         self.__open_element('texture_instance name="{0}_inst" texture="{1}"'.format(texture_name, texture_name))
         self.__emit_parameter("addressing_mode", mode)
         self.__emit_parameter("filtering_mode", "bilinear")
+        self.__emit_parameter("alpha_mode", "detect")
```
Comment (dictoon) [inline projectwriter.py]: This is completely correct and probably the best thing to do, but it's worth noting that detecting the mode of an alpha texture is not free:

For RGBA textures, appleseed needs to iterate over the texels of the texture until either one is found with an alpha channel < 1 in which case the mode will be `"alpha_channel"`, or until all pixels have been visited and they all have alpha == 1 in which case the mode will be `"luminance"`.

Of course no scan is done at all for RGB textures.

**Comment by jdent02** [comment]
I could add an alpha mode selector to the texture selectors.  Or would that be more effort than it’s worth?

Sent from my iPhone

On Jan 26, 2018, at 8:02 AM, François Beaune <notifications@github.com<mailto:notifications@github.com>> wrote:


@dictoon commented on this pull request.

________________________________

In projectwriter.py<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fappleseedhq%2Fblenderseed%2Fpull%2F172%23discussion_r164118294&data=02%7C01%7C%7Cf75d12c03afd48ce8f6708d564c58054%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636525721774056961&sdata=tiNXi%2BHAp5KDjnHz7mwKPVdPcqPJdVy0VQOT7l0wHjE%3D&reserved=0>:

> @@ -2221,6 +2221,7 @@ def __emit_texture_instance(self, texture, texture_name, bump_bool, node=None, m
         self.__open_element('texture_instance name="{0}_inst" texture="{1}"'.format(texture_name, texture_name))
         self.__emit_parameter("addressing_mode", mode)
         self.__emit_parameter("filtering_mode", "bilinear")
+        self.__emit_parameter("alpha_mode", "detect")


This is completely correct and probably the best thing to do, but it's worth noting that detecting the mode of an alpha texture is not free:

For RGBA textures, appleseed needs to iterate over the texels of the texture until either one is found with an alpha channel < 1 in which case the mode will be "alpha_channel", or until all pixels have been visited and they all have alpha == 1 in which case the mode will be "luminance".

Of course no scan is done at all for RGB textures.

**Comment by dictoon** [comment]
I think it would make sense for completeness' sake but it's certainly not a priority. I'll add a ticket and we can schedule that for later or let a student take it.

**Comment by dictoon** [comment]
Here's a ticket: https://github.com/appleseedhq/blenderseed/issues/173

**Pull request merged**